lang/java/reef-utils/src/main/java/org/apache/reef/util/SimpleCondition.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.reef.util;
+
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Manages Java lock and condition objects to create a simplified
+ * condition variable interface.
+ */
+public final class SimpleCondition {
+  private final Lock lockVar = new ReentrantLock();
+  private final Condition conditionVar = lockVar.newCondition();
+  private final long timeoutPeriod;
+  private final TimeUnit timeoutUnits;
+  private static final long DEFAULT_TIMEOUT = 10;
+
+  /**
+   * Default constructor which initializes timeout period to 10 seconds.
+   */
+  public SimpleCondition() {
+    this(DEFAULT_TIMEOUT, TimeUnit.SECONDS);
```

Comment:
Same as complex condition: maybe we should wait indefinitely by default

Reply:
Done